### ISSUE TITLE
Final Polish: Pass Config to `RegimeModelTrainer` Constructor to Eliminate Warnings

### ISSUE THREAD

**Issue by SefaGH** [SefaGH/bearish-alpha-bot#393]
### 🎯 Amaç

Bu son görev, `train-models.yml` workflow'unun loglarındaki son `WARNING` mesajını (`Config not provided`) ortadan kaldırarak ve `RegimeModelTrainer` sınıfının mimari olarak doğru çalışmasını sağlayarak GEMMA entegrasyonunu tamamlamayı hedefler. `train_all_models.py` script'i, `RegimeModelTrainer` sınıfını başlatırken, ona merkezi `gemma_config` nesnesini constructor üzerinden iletecek şekilde onarılacaktır.

#### 1. Konfigürasyon Aktarımını Düzelt (`scripts/train_all_models.py`)

**Sorun:** `train_gemma_model` fonksiyonu, `RegimeModelTrainer`'ı `trainer = RegimeModelTrainer()` şeklinde boş başlatıyor. Bu, trainer'ın `config.example.yaml`'daki optimize edilmiş `gemma` ayarlarını alamamasına neden oluyor.

**Çözüm:** Trainer'ı, `gemma` konfigürasyonunu doğrudan alacak şekilde başlat.

- **Dosyayı Aç:** `scripts/train_all_models.py` dosyasını aç.
- **`train_gemma_model` Fonksiyonunu Bul:** Bu fonksiyonun içindeki `RegimeModelTrainer`'ın başlatıldığı satırı bul.
- **Başlatma Satırını Güncelle:**

  ```python
  # scripts/train_all_models.py -> train_gemma_model fonksiyonu içinde
  
  # ESKİ ve HATALI YAPI:
  # trainer = RegimeModelTrainer() 
  
  # YENİ ve MİMARİ OLARAK DOĞRU YAPI:
  gemma_config = config.get('gemma', {})
  if not gemma_config:
      raise ValueError("Konfigürasyonda 'gemma' bölümü bulunamadı.")
  
  # Trainer'ı bu spesifik konfigürasyon ile başlat.
  trainer = RegimeModelTrainer(config=gemma_config)
  
  logger.info("Merkezi model eğitici (RegimeModelTrainer) doğru konfigürasyon ile başlatıldı.")
  ```

#### 2. `RegimeModelTrainer` İçindeki Geçici Çözümü Kaldır (`src/ml/model_trainer.py`)

- **Dosyayı Aç:** `src/ml/model_trainer.py` dosyasını aç.
- **`__init__` Metodunu Kontrol Et:** Bu metodun, `config` parametresini alıp `self.config = config` şeklinde bir sınıf değişkenine atadığından emin ol.
- **`train_and_evaluate` Metodunu Temizle:** Copilot'un `config`'i yeniden okumak için eklediği geçici kodları kaldır. Metodun, doğrudan `self.config`'i kullanması gerekir.

### ✅ Başarı Kriterleri

- `train-models.yml` workflow'u çalıştırıldığında, loglarda artık **`WARNING - Config not provided`** uyarısı **kesinlikle görülmemelidir.**
- Loglarda, `RegimeModelTrainer`'ın doğru konfigürasyonla başlatıldığına dair bir `INFO` mesajı yer almalıdır.
- En önemlisi, modelin test doğruluk (`Accuracy`) oranının, **tüm optimize edilmiş hiperparametreler** (learning_rate, batch_size, dropout vb.) artık doğru kullanıldığı için, hedeflenen **~%78** seviyesine önemli ölçüde yaklaşması beklenir.

Bu son görev tamamlandığında, `train-models.yml` üretim hattı, sadece çalışan değil, aynı zamanda mimari olarak temiz, verimli ve en yüksek performansı üreten nihai haline kavuşmuş olacaktır. 🎉
